### PR TITLE
feat: feature/icky 802 subscription on pdp page

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 1]
```diff
@@ -349,14 +418,41 @@ const ProductDetailTemplate = (props: ProductDetailTemplateProps) => {
...
+            <FulfillmentOptions
+              fulfillmentOptions={fulfillmentOptions}
+              selected={selectedFulfillmentOption?.method}
+              onFulfillmentOptionChange={(value: string) => handleFulfillmentOptionChange(value)}
+              onStoreSetOrUpdate={() => handleProductPickupLocation()}
+            />
+          )}
         </Box>
 
         <Box pt={2} display="flex" sx={{ justifyContent: 'space-between' }}>
```
Comment (kibo-sushant) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:446]: Unable to test using Storybook
Unit Testcases are missing

**Review by kibo-sushant — CHANGES_REQUESTED** [review]

File: lib/getters/productGetters.ts [hunk 2]
```diff
@@ -196,6 +206,7 @@ const getProductDetails = (product: ProductCustom) => {
     variationProductCode: getVariationProductCodeOrProductCode(product),
     fulfillmentMethod: getSelectedFulfillmentOption(product),
     productPrice: getPrice(product),
+    productSubscriptionPrice: getSubscriptionPrice(subscriptionPrice),
```
Comment (kibo-sushant) [inline lib/getters/productGetters.ts]: Update the unit testcases accordingly.

File: lib/getters/subscriptionGetters.ts [hunk 3]
```diff
@@ -51,4 +79,6 @@ export const subscriptionGetters = {
   getSubscriptionFrequency,
   getSubscriptionDetails,
   getFrequencyValues,
+  isSubscriptionModeAvailable,
+  getSubscriptionFrequencyUnit,
```
Comment (kibo-sushant) [inline lib/getters/subscriptionGetters.ts]: Update the unit testcases accordingly.

**Review by kibo-sushant — CHANGES_REQUESTED** [review]

File: lib/getters/subscriptionGetters.ts [hunk 4]
```diff
@@ -35,11 +37,37 @@ const getSubscriptionDetails = (subscription: any) => {
...
+  if (!product) return false
+
+  return (
+    product?.properties?.some(
+      (property) => property?.attributeFQN === ProductAttribute.SUBSCRIPTION_Mode
+    ) || false
+  )
+}
+
+const getSubscriptionFrequencyUnit = (selectedFrequency: string) => {
```
Comment (kibo-sushant) [inline lib/getters/subscriptionGetters.ts]: name is not meaningful (we are returning both unit and value)

File: components/page-templates/ProductDetail/ProductDetailTemplate.spec.tsx [hunk 5]
```diff
@@ -60,6 +61,43 @@ jest.mock(
...
+      <input
+        type="radio"
+        name="onetime_purchase"
+        value={PurchaseTypes.ONETIMEPURCHASE}
+        onClick={() => handlePurchaseTypeSelection({ option: PurchaseTypes.ONETIMEPURCHASE })}
+      />
+    </div>
+  ),
+}))
+
```
Comment (kibo-sushant) [inline components/page-templates/ProductDetail/ProductDetailTemplate.spec.tsx]: alongwith handlePurchaseTypeSelection also pass selected prop and verify it
Not mocking KiboSelect, Price, QuantitySelector ?
Unit Test is not covering all the scenarios of the code

Comment (kibo-sushant) [inline components/page-templates/ProductDetail/ProductDetailTemplate.spec.tsx]: alongwith handlePurchaseTypeSelection also pass selected prop and verify it
Not mocking KiboSelect, Price, QuantitySelector ?
Unit Test is not covering all the scenarios of the code

**Review by kibo-sushant — CHANGES_REQUESTED** [review]

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 1 above)
Comment (Kibo-TriptiV) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx:446]: Done

File: lib/getters/productGetters.ts (hunk 2 above)
Comment (Kibo-TriptiV) [reply lib/getters/productGetters.ts]: Done

File: lib/getters/subscriptionGetters.ts (hunk 3 above)
Comment (Kibo-TriptiV) [reply lib/getters/subscriptionGetters.ts]: Done

File: lib/getters/subscriptionGetters.ts (hunk 4 above)
Comment (Kibo-TriptiV) [reply lib/getters/subscriptionGetters.ts]: Done

File: components/page-templates/ProductDetail/ProductDetailTemplate.spec.tsx (hunk 5 above)
Comment (Kibo-TriptiV) [reply components/page-templates/ProductDetail/ProductDetailTemplate.spec.tsx]: Not required to mock for subscription tests, I have covered unit test for subscription now. Need to refactor existing tests (may be we can have separate ticket).

Comment (Kibo-TriptiV) [reply components/page-templates/ProductDetail/ProductDetailTemplate.spec.tsx]: Done

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 6]
```diff
@@ -81,6 +91,16 @@ const ProductDetailTemplate = (props: ProductDetailTemplateProps) => {
   const { getProductLink } = uiHelpers()
   const { product, breadcrumbs = [], isQuickViewModal = false, children } = props
   const { t } = useTranslation('common')
+
+  const [purchaseType, setPurchaseType] = useState<string>(PurchaseTypes.ONETIMEPURCHASE)
+  const [selectedFrequency, setSelectedFrequency] = useState<string>('')
+
+  const isSubscriptionModeAvailable = subscriptionGetters.isSubscriptionModeAvailable(product)
+  const { data: subscriptionPrice } = useProductPriceQueries(
```
Comment (kibo-kevinwatts) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx]: we always want to use the price returned from the `useProductPriceQuery` not just when its subscription price. we should rename and make changes for hook to always load

File: hooks/queries/useProductPriceQueries/useProductPriceQueries.ts [hunk 7]
```diff
@@ -55,7 +56,8 @@ export const useProductPriceQueries = (
     productKeys.productParams(productCode, useSubscriptionPricing),
     () => fetchProductPrice(productCode, useSubscriptionPricing),
     {
-      refetchOnWindowFocus: false,
+      enabled: !!(productCode && useSubscriptionPricing),
```
Comment (kibo-kevinwatts) [inline hooks/queries/useProductPriceQueries/useProductPriceQueries.ts]: as mentioned in pdp template comment, don't disable based on `useSubscriptionPricing`

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

**Review by kibo-sushant — APPROVED** [review]

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 6 above)
Comment (Kibo-TriptiV) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx]: Done

File: hooks/queries/useProductPriceQueries/useProductPriceQueries.ts (hunk 7 above)
Comment (Kibo-TriptiV) [reply hooks/queries/useProductPriceQueries/useProductPriceQueries.ts]: Done

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**